core/dbt/task/init.py

```diff
@@ -302,26 +302,33 @@ def run(self):
             self.create_profile_from_target(
                 adapter, profile_name=profile_name
             )
-        else:
-            # When dbt init is run outside of an existing project,
-            # create a new project and set up the user's profile.
+            return
+
+        # When dbt init is run outside of an existing project,
+        # create a new project and set up the user's profile.
+        project_name = self.args.project_name
+        if project_name is None:
+            # If project name is not provided,
+            # ask the user which project name they'd like to use.
             project_name = click.prompt("What is the desired project name?")
-            project_path = Path(project_name)
-            if project_path.exists():
-                fire_event(ProjectNameAlreadyExists(name=project_name))
-                return
-
-            self.copy_starter_repo(project_name)
-            os.chdir(project_name)
-            with open("dbt_project.yml", "r+") as f:
-                content = f"{f.read()}".format(
-                    project_name=project_name,
-                    profile_name=project_name
-                )
-                f.seek(0)
-                f.write(content)
-                f.truncate()
+        project_path = Path(project_name)
+        if project_path.exists():
+            fire_event(ProjectNameAlreadyExists(name=project_name))
+            return
+
+        self.copy_starter_repo(project_name)
+        os.chdir(project_name)
+        with open("dbt_project.yml", "r+") as f:
+            content = f"{f.read()}".format(
+                project_name=project_name,
+                profile_name=project_name
+            )
+            f.seek(0)
+            f.write(content)
+            f.truncate()
```

Comment:
I think it would be great, in the future, to add another log/event here to explain what's happening. That way:
```
$ dbt init my_new_project -s
11:54:50 | [ info  ] | Running with dbt=1.0.0-b2
11:54:51 | [ info  ] | Created starter project named 'my_new_project'
```
That can come in another future pass